source/fundamentals/crud/run-command.txt

```diff
@@ -0,0 +1,159 @@
+.. _golang-run-command:
+
+=============
+Run a Command
+=============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to run a raw **database command** with the
+{+driver-short+}. You can use database commands to perform a variety of
+administrative and diagnostic tasks, such as fetching server statistics,
+initializing a replica set, or running an aggregation pipeline.
+
+Compose a Command
+-----------------
+
+The {+driver-short+} provides two methods to run a database command. The
+``RunCommand()`` method executes the given command against the database
+and returns a ``SingleResult`` type. The ``RunCommandCursor()`` method
+executes the given command against the database and parses the response
+as a cursor.
+
+.. note:: Read Preference
+
+   These methods do not obey the read preference set for the database.
+   You can set a read preference for a database command by passing a
+   options to either method:
+   
+   .. code-block:: go
+
+      opts := options.RunCmd().SetReadPreference(readpref.Primary())
+      err := db.RunCommand(context.TODO(), command, opts).Decode(&result)
+   
+   For more information on
+   read preference options, see the :ref:`golang-write-read-pref`
+   fundamentals page.
+
+Each method takes a Context and command document. The
+command document must be an order-preserving type such as ``bson.D``.
+The following code shows how you can run the ``listCollections`` command
+on a database with the ``RunCommandCursor()`` method:
+
+.. code-block:: go
+
+   command := bson.D{{"listCollections", 1}}   
+   cursor, err := db.RunCommandCursor(context.TODO(), command)
+
+For a full list of database commands, see the :ref:`Additional Information <addl-info-runcommand>`.
+
+Response
+--------
+
+Each method returns a document or a cursor containing documents with the
+following fields:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - Field
+     - Description
+
+   * - <command result>
+     - Provides fields specific to the database command.
```

Comment:
Issue: A little vague

Suggestion: Provide examples of some fields that commands might return (like `n` for count).